Eliminate LearningTest directory naming requirement from kht_test

bruno-at-orange opened the pull request:
Modification essentiellement dans _kht_utils.py:

    check_home_dir:
        un repertoire est un home dir de LearningTest s'il contient au moins un des sous-reprtoire de test des outils
        au lieu de simplement comparer le nom à "LearningTest"
    on passe désormais systematiquement par la methode check_home_dir, avec un chemin en parametre
        ajout de la fonction utilitaire parent_dir_path, soeur de la fonction parent_dir_name

Tests avec des repertoires de nom quelconques, valides ou non valides

Plus de nombreuse corrections de typos dans les commentaire, dans un commit a part.

cherry pick from dev

marcboulle reviewed (approved):
J'ai tout comparé avec la version de dev: c'est ok

Eventuellement, juste un commentaire erronée à corriger
- kht_test.py, ligne 84
  "# Erreur si plusieurs repertoires des binaires non trouve"
  => "# Erreur si plusieurs repertoires des binaires trouves"

Cette PR ferme en fait l'issue https://github.com/KhiopsML/khiops/issues/249, qu'il faudra fermer manuellement.

bruno-at-orange committed:
Eliminate LearningTest directory naming requirement from kht_test

Modification essentiellement dans _kht_utils.py:

    check_home_dir:
        un repertoire est un home dir de LearningTest s'il contient au moins un des sous-reprtoire de test des outils
        au lieu de simplement comparer le nom à "LearningTest"
    on passe désormais systematiquement par la methode check_home_dir, avec un chemin en parametre
        ajout de la fonction utilitaire parent_dir_path, soeur de la fonction parent_dir_name

Tests avec des repertoires de nom quelconques, valides ou non valides

Plus de nombreuse corrections de typos dans les commentaire, dans un commit a part.

cherry pick from dev
